firebase-perf/src/main/java/com/google/firebase/perf/metrics/AppStartTrace.java

```diff
@@ -228,12 +274,30 @@ public synchronized void onActivityResumed(Activity activity) {
     // Log the app start trace in a non-main thread.
     executorService.execute(this::logAppStartTrace);
 
-    if (isRegisteredForLifecycleCallbacks) {
+    if (!isExperimentTTIDEnabled && isRegisteredForLifecycleCallbacks) {
       // After AppStart trace is logged, we can unregister this callback.
       unregisterActivityLifecycleCallbacks();
     }
   }
 
+  private void logColdStart(Timer start, Timer end, PerfSession session) {
+    TraceMetric.Builder metric =
+        TraceMetric.newBuilder()
+            .setName("_experiment_app_start_ttid")
```

Comment:
nit: define the name as a constant Constants.TraceNames.

Reply:
This code is for dark-launch, it will be removed next release, so I didn't add a new official constant to Constants.java. I will definitely do it for the GA launch.